### DISCUSSION TITLE
Pass custom config to custom UI

### DISCUSSION THREAD

**Discussion by DiesIrae** [keystonejs/keystone#7483] [Questions] [answered]
I created a custom UI for a field, via `ui: { views: require.resolve(...) }`, used in item & create views.
This custom component UI has front-end configurations I want to pass to. How can I pass them to my Field component?

**Comment by raveling** [comment, marked as answer]
@DiesIrae currently the only way to do this is to create a [custom field](https://keystonejs.com/docs/guides/custom-fields#custom-fields) type. We have an [example](https://github.com/keystonejs/keystone/tree/main/examples/custom-field) you can try in local to learn more.

**Reply by DiesIrae** [reply]
ok, I see. It adds a lot of boilerplate just for a small use case, but at least it can be done. Thanks Ronald!

**Reply by gautamsi** [reply]
I agree, We should have a way to define the custom UI options for the fields. For ypescript we can use generic if needed like this (ideally)
```
myAwesomeTextField = text<CustomUIOPtion>({...fieldOptions, ui: {config: {...customFieldOptions}}})
```
this way the type safety will also be there and we should be able to get this back in `Controller`, `Field` or `Cell` to use those config
